wasm/js/worker.js

```diff
@@ -1,4 +1,4 @@
-import wasm from "../Cargo.toml";
+import wasm from "../dist/wasm.js";
```

Comment:
Both `index.js` and `worker.js` import the `Cargo.toml` file.

Because they're importing the `Cargo.toml` file, Rollup will then compile the Rust Wasm code.

Normally Rollup would just build the Rust Wasm one time, and share it with the `index.js` and `worker.js` files.

However, it's using multiple builds, a separate build for `index.js` and `worker.js`.

Because the builds are separate, that means Rollup will not share anything between the builds.

That means Rollup will compile the Rust Wasm code twice, and it will create 2 separate `.wasm` files.

Since the Rust Wasm code takes approximately 10 eternities to compile, since it's compiled twice that means it will now take 20 eternities.

So my solution for that is to build the Rust Wasm in a separate Rollup build, which will output the Rust Wasm compiled code to `dist/wasm.js` and `dist/assets/aleo_wasm.wasm`.

After Rollup puts the compiled Rust Wasm into `dist/wasm.js`, it then builds the `index.js` and `worker.js` files separately.

And the `index.js` and `worker.js` files can then import the `dist/wasm.js` file (which contains the compiled Rust Wasm code).

So that way it only needs to build the Rust Wasm one time, and it can reuse the same `.wasm` for both `index.js` and `worker.js`.

Reply:
Got it, does that mean users can start multiple workers with `new Worker(new URL("worker.js", import.meta.url), { type: "module" });` as it's needed?

Reply:
Sure, that's possible, though it won't actually increase parallelism, since the Rayon threads should be fully saturating the CPU.

So instead it would probably be better for the user to just use `Promise.all` for concurrency within a single Worker:

```js
const [a, b] = await Promise.all([
    localProgramExecution(),
    localProgramExecution(),
]);
```

That should also run both programs at the same time, but it's able to share a single Wasm instance + memory, and share a single Rayon thread pool, and it's able to share the private keys.

So this should be better than spawning multiple Workers.